### PR TITLE
Clean up NumPyro DM mass-selection API

### LINKED ISSUE #31: Clean up NumPyro DM mass-selection API before v0.1.0
Follow-up to #29. The numerical/autodiff blocker is resolved, but the public NumPyro Jeans API still exposes `use_analytic_dm: Optional[bool] = None`, and the model-aware `auto` policy is encoded via `_default_enclosed_mass_method` rather than an explicit autodiff-safety capability.

Please implement this cleanup before v0.1.0.

## Required API change

Replace `use_analytic_dm` on all three public/internal Jeans solver entry points with a method-style option:

```python
dm_mass_method: str = "auto"
```

Supported values must be exactly:

- `"auto"`
- `"analytic"`
- `"numeric"`

Apply consistently to:

- `DSphModel.sigmalos2`
- `DSphModel.sigmalos2_kernel`
- `DSphModel.sigmalos2_abel`

There were no pre-v0.1 repository callers that require preserving `use_analytic_dm`, so remove it rather than keeping a three-state compatibility shim unless a current-main caller proves otherwise.

## Capability-driven auto selection

Make `DMModel.enclosed_mass(method="auto")` capability-driven rather than relying on a model-specific default-method string. Prefer an explicit semantic capability such as:

```python
analytic_enclosed_mass_autodiff_safe = False
```

on `DMModel`, overridden to `True` for `NFWModel`. General `ZhaoModel` should remain `False` because its analytic `jax.scipy.special.betainc` path is not autodiff-safe with respect to Zhao shape parameters `(a, b, g)`.

Then resolve `method="auto"` as:

- analytic only if an analytic enclosed-mass implementation exists **and** the model declares that analytic path autodiff-safe;
- otherwise numeric.

Keep explicit `method="analytic"` and `method="numeric"` overrides available.

## Validation / behavior

Preserve the intended behavior introduced in #29:

- NFW + auto -> analytic mass;
- general Zhao + auto -> numeric mass;
- explicit Zhao analytic remains available for value/regression work when shape-parameter gradients are not required.

Validate method strings consistently and raise a clear `ValueError` for invalid values.

## Tests

Update/add tests so they cover at least:

1. NFW `method="auto"` selects/matches analytic.
2. Zhao `method="auto"` selects/matches numeric.
3. Explicit Zhao numeric vs analytic value comparison remains.
4. Default `DSphModel.sigmalos2` Zhao path still has a finite `jax.grad` through at least one Zhao shape parameter.
5. `dm_mass_method="analytic"` and `dm_mass_method="numeric"` explicitly propagate through kernel and Abel paths as intended.
6. Invalid `dm_mass_method` is rejected.
7. No active source/tests/docs still use `use_analytic_dm`.

Update cache/JIT static keys if necessary so `dm_mass_method` participates correctly in cached solver behavior.

## Documentation

Update README/docstrings from `use_analytic_dm` to `dm_mass_method`, while preserving the explanation that Zhao's analytic `betainc` expression is not JAX-autodiff-safe in its shape parameters and that `auto` is the NUTS-safe default.

## Final checks

Start from current `main`, keep the scope limited to this API cleanup, run focused tests and the full suite, and report exact results. Also run/search for leftover `use_analytic_dm` references before finalizing.

@copilot Please implement this issue on a branch and open a PR for review.

### PR DESCRIPTION
**Pull request by gomeshun**
Follow-up to #29 and implementation target for #31.

## What changed

- Replaced `use_analytic_dm: Optional[bool]` with `dm_mass_method="auto"|"analytic"|"numeric"` on `DSphModel.sigmalos2`, `sigmalos2_kernel`, and `sigmalos2_abel`.
- Made `DMModel.enclosed_mass(method="auto")` capability-driven through `analytic_enclosed_mass_autodiff_safe`.
  - NFW: analytic enclosed mass is autodiff-safe, so `auto` selects analytic.
  - General Zhao: analytic `betainc` shape-parameter gradients are not JAX-safe, so `auto` selects numeric.
- Preserved explicit analytic/numeric overrides.
- Added consistent validation for invalid `dm_mass_method` values.
- Updated README and existing regression/benchmark calls to the new API.
- Added API contract tests for the public signatures and auto-selection behavior.
- Kept the existing Zhao `jax.grad` regression as the single differentiability test (duplicate follow-up test removed after review).

## Validation

Before the final test-deduplication commit, focused tests reported `40 passed, 3 skipped, 9 warnings, 37 subtests passed` and the full suite reported `163 passed, 3 skipped, 37 subtests passed`. The final commit only removes the duplicate gradient regression; GitHub Actions is rerunning on that head.

Closes #31

### CODE REVIEW

**Comment by gomeshun** [comment]
@copilot Please implement the follow-up now. The regression test file on this branch defines the intended API contract; use issue #31 as the full specification. Update `src/jeanspy/model_numpyro.py`, README/docstrings, existing tests/benchmarks/cache keys as needed so that:

1. `sigmalos2`, `sigmalos2_kernel`, and `sigmalos2_abel` expose `dm_mass_method: str = "auto"` and remove `use_analytic_dm`.
2. `DMModel.enclosed_mass(method="auto")` is capability-driven with an explicit semantic flag such as `analytic_enclosed_mass_autodiff_safe = False`; NFW overrides it to `True`, Zhao remains `False`.
3. `auto` remains NFW -> analytic and Zhao -> numeric; explicit analytic/numeric overrides remain.
4. invalid method names raise clear `ValueError`s.
5. the existing finite `jax.grad` Zhao regression remains green.
6. all active `use_analytic_dm` references are removed and docs updated.

Please run the focused tests plus the full suite, report exact results, and push the implementation to this PR branch. Do not merge; leave it for review.

File: tests/test_dm_mass_method_api.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            u_max=300.0,
+            dm_mass_method="not-a-method",
+        )
+
+
+def test_default_zhao_jeans_path_remains_differentiable():
+    dsph = _zhao_dsph()
+    params = _zhao_params()
+    R = jnp.asarray([20.0, 100.0], dtype=jnp.float32)
+
```
[the remark this answers is not in the document]
Comment (gomeshun) [reply tests/test_dm_mass_method_api.py]: Agreed. I removed the duplicate `jax.grad` regression from `tests/test_dm_mass_method_api.py`; the existing `tests/test_model_numpyro.py::test_zhao_default_sigmalos2_path_has_finite_shape_gradient` remains the single differentiability regression. This keeps the API contract tests lightweight while preserving the NUTS-safety check.

**Review by gomeshun — COMMENTED** [review]
Implementation review on current head: the capability-driven auto policy, public `dm_mass_method` API, explicit overrides, validation, documentation, and JIT cache key all look coherent. I addressed the only existing review thread by removing the duplicate Zhao `jax.grad` regression while retaining the original differentiability test. Please re-review the latest head for any remaining correctness/API issues.

**Comment by gomeshun** [comment]
@copilot Please re-review the latest head `dfaf6f7`. The implementation is now present, the stale draft description is updated, and the duplicate Zhao `jax.grad` contract test flagged in the previous review has been removed while retaining the existing differentiability regression in `tests/test_model_numpyro.py`. Please check for any remaining correctness/API issues only; do not modify or merge.
